### PR TITLE
Enumerables

### PR DESCRIPTION
**Pull request by Tommymarvel**
In the branch I have implemented the following instructions:
- Create a class `MyList` that has an instance variable `@list`.
- In `MyList` implement a method `#each` that yields successive members of `@list` and uses the `MyEnumerable` module.
- Create a module `MyEnumerable` that implements the following methods (they should have the same functionality as methods in [Enumerable](https://ruby-doc.org/core-3.0.0/Enumerable.html)):
  - `#all?`
  - `#any?`
  - `#filter`
- Each class and module should have a separate .rb file.

### CODE REVIEW

File: mylist.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative 'my_enumerable'
+
+class MyList
+  include MyEnumerable
+  def initialize(*initial)
```
Comment (Ismailco) [inline mylist.rb]: - [ ] MyList class should have an instance variable `@list`, kindly consider implementing this inside the `initialize` method so the `MyList` class has a list

File: mylist.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    @initial = initial
+  end
+
+  def each(&block)
```
Comment (Ismailco) [inline mylist.rb]: - [ ] As the requirement mentioned you should implement the `#each` method that yields all members of `@list` in succession without the use of the ruby `.each` method

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @Tommymarvel, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️  No linter errors
✔️ Used GitHub Flow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**You can also consider personalizing your README file so it documents your work, kindly add the project name, description, and what to do next... so your documentation looks accurate and professional**

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: mylist.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+require_relative 'my_enumerable'
+
+class MyList
+  include MyEnumerable
+  def initialize(*list)
+    @list = list
+  end
+
+  def each(&block)
+    @list.each(&block)
```
Comment (m05e5) [inline mylist.rb]: - [ ] Hi as required and as mentioned by the previous reviewer, the `each` method should yield all members of @list in succession, but you shouldn't use the ruby `each` keyword method. you have to design it yourself

**Review by m05e5 — CHANGES_REQUESTED** [review]
Hi @Tommymarvel , 

Good job so far. !
There are some issues that you still need to work on to go to the next project but you are almost there!
mario_ladders


## Highlights


## Required Changes ♻️
* You really did a great job but there are some features you need to improve

- [ ] Please check the comments below 

_I'm sure you can do it just keep it up_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_@m05e5_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
Hi @Tommymarvel, 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**